test/tests/api-cit/v2_0/nodes_tests.py

```diff
@@ -464,39 +452,63 @@ def test_node_get_obm_by_node_id(self):
         rsp = self.__client.last_response
         nodes = loads(rsp.data)
         self.assertEqual(200, rsp.status, msg=rsp.status)
+        test_obm = {
+            'config': {
+                'host': '1.2.3.4',
+                'user': 'username',
+                'password': 'password'
+            },
+            'service': 'noop-obm-service'
+        }
+        # get the first test compute node id, to get a 404, we need the payload data set up
+        # with a valid node id. Otherwise, a 400 will be returned if bad payload.
         for n in nodes:
             if n.get('name') == 'test_compute_node':
-                try:
-                    Api().nodes_put_obms_by_node_id(identifier='invalid_ID', body=self.__test_obm)
-                    self.fail(msg='did not raise exception')
-                except ApiException as e:
-                    self.assertEqual(404, e.status, msg='unexpected response {0}, expected 404'.format(e.status))
+                test_obm["nodeId"] = str(n.get('id'))
+                found_node = True
+                break
+        if found_node:
+            try:
+                Api().nodes_put_obms_by_node_id(identifier='invalid_ID', body=test_obm)
+                self.fail(msg='did not raise exception')
+            except ApiException as e:
+                self.assertEqual(404, e.status, msg='unexpected response {0}, expected 404'.format(e.status))
+        else:
+            self.fail(msg='No test compute node available for try invalid_ID')
 
-    # @test(groups=['node_get_obm_invalid'], depends_on_groups=['node_put_obm_invalid'])
     @depends(after='test_node_put_obm_invalid_node_id')
     def test_node_get_obm_invalid_node_id(self):
         # Testing that PUT:/api/2.0/:id/obm returns 404 with invalid node ID
-        Api().nodes_get_all()
-        rsp = self.__client.last_response
-        nodes = loads(rsp.data)
-        self.assertEqual(200, rsp.status, msg=rsp.status)
-        for n in nodes:
-            if n.get('name') == 'test_compute_node':
-                try:
-                    Api().nodes_get_obms_by_node_id(identifier='invalid_ID')
-                    self.fail(msg='did not raise exception')
-                except ApiException as e:
-                    self.assertEqual(404, e.status, msg='unexpected response {0}, expected 404'.format(e.status))
+        try:
+            Api().nodes_get_obms_by_node_id(identifier='invalid_ID')
+            self.fail(msg='did not raise exception')
+        except ApiException as e:
+            self.assertEqual(404, e.status, msg='unexpected response {0}, expected 404'.format(e.status))
 
     @depends(after='test_node_get_obm_invalid_node_id')
     def test_clean_up_test_nodes(self):
         # Clean up the added test nodes
         self.assertEqual(0, self.delete_temp_nodes(), msg="Failed to clean up test nodes from script")
+
+    @classmethod
+    def clear_nodes(self):
+        # """ Clear the temp created nodes if any exist """
```

Comment:
Yes, it does delete it from the node_list via  Api().nodes_del_by_id(identifier=uuid).